packages/apps/shopify-api/lib/auth/scopes/index.ts

```diff
@@ -6,6 +6,7 @@ class AuthScopes {
 
   private compressedScopes: Set<string>;
   private expandedScopes: Set<string>;
+  private originalScopes: Set<string>;
```

Comment:
Should we consider how the other methods are implemented as well?

The current behaviour considers AuthScopes(['read_products']) equivalent to AuthScopes(['read_products', 'write_products']). Since we are encouraging developers to use this class, I wonder if this behaviour is consistent with the changes proposed in this PR / as part of the Optional scopes feature?

Reply:
Ideally, we would want to keep the scope of this change tight and not introduce breaking changes (by changing the default behaviour). However, I wonder about the developer use cases. Would they want to compare and check inclusion with respect to the exact set of scopes or the [expanded](https://github.com/Shopify/shopify-app-js/blob/9733952bfa79f65bde80df7b3c718ac2f67047c4/packages/apps/shopify-api/lib/auth/scopes/index.ts#L52) set of scopes? What about the other methods?


i.e.
```
required: ['read_products']
optional: ['write_products']
granted: ['read_products']
```

1. Developer queries for optional scopes and instantiates:
```
requiredScopes = AuthScope(['read_products']);
optionalScopes = AuthScope(['write_products']);
grantedScopes = AuthScope(['read_products']);
```
2. Developer wants to check whether the optional scopes have been granted:
```
grantedScopes.has(optionalScopes);
grantedScopes.equals(optionalScopes);
```

* Is this a realistic use case?
* **Is current implementation behaving as expected in these scenarios?**


cc @kbav